### DISCUSSION TITLE
@slidy/core - prevent scroll axis crossing on touch/trackpad

### DISCUSSION THREAD

**Discussion by inspiredlabs** [Valexr/Slidy#57] [Q&A] [answered]
With the new version of `slidy/core`, will it be possible to pass in `sensitivity` in `px`? 

In the previous version of slidy [you could pass in `sensity`](https://svelte.dev/repl/5c149e1aeef242bd889a7ecf115c02b5?version=3.46.4), which worked well enough, eg:

```
options: {
  sensity: 6.25, // lower number is more sensitive
}
```

Ideally you would pass in `96` to the vertical content wrapper to prevent a horizontal slide. This is especially useful with a mousewheel, or trackpad.

Thank you for Slidy btw. One day I hope it will compete with [Flickity](https://flickity.metafizzy.co/extras.html)!

**Comment by Valexr** [comment]
Thanks for regards @inspiredlabs, maybe 😁, but in bundle size `@slidy/core` wins of all carousel scripts now🤓

`sensity` props excluded from version 3. In core you have `gravity` & `duration` for kinetic tuning.
See in [example](https://svelte.dev/repl/8edad715f4054a20ac9b43af28b17083)
But haven't props for preventing from scroll crossing... okey will see ) maybe it will be default defending behaviour...🤔

And version @slidy/core without `indexer()`, `scroller()` function - this variables will passing through event `move` on scrollable/target node in function `slidy(node, options)`  in progress🚀

And version `@slidy/svelte@3.0` in progress now. All styles customisation through css vars & more...
Stay tune🤓

**Comment by Valexr** [comment]
@inspiredlabs & 1 more 😁in now development version `@slidy/core` allready preventing mouse wheel from scroll crossing & wheel sliding possible only with `shift` key

**Comment by Valexr** [comment]
@inspiredlabs fixed - now on wheel event with `vertical: false` preventing horizontal page scroll event (nav of browser history), but have free vertical scroll page. In `vertical: true` mode not preventing any default browser behavior. 
And with `e.shiftKey` both axis preventing default & Slidy sliding by indexes🤓

**Comment by inspiredlabs** [comment]
Brilliant, that was quick!

Please can you walk me through getting it to work on [this](https://github.com/inspiredlabs/offline/commit/cd4d36be3e592cdf93b3bdb633882575fd12781a) repo?

Here's what I've done:

- `npm i @slidy/core` in the project root following: https://www.npmjs.com/package/@slidy/core
- passed in `vertical: false,` to `<ul use:slidy={{ ... }}`.
- scrolled using the trackpad, see *screenshot*
- pushed it to [this](https://github.com/inspiredlabs/offline/commit/cd4d36be3e592cdf93b3bdb633882575fd12781a) repo.

![Screenshot 2022-02-25 at 12 14 49](https://user-images.githubusercontent.com/5474496/155706960-f182b21a-9f4d-4b76-a098-93e1e4875a5e.jpg)

**Comment by Valexr** [comment]
Fixed but not published - i'm working on global changes in new version 🤓sorry
I'm see the [demo](https://offline-phi.vercel.app/) - all working good  (as expectedly) on desktop & mobile ...🤷🏻‍♂️

What's yours problem?

**Comment by Valexr** [comment]
https://user-images.githubusercontent.com/5656366/155711495-9372208d-2cd9-4ece-bdeb-220cf1289d8f.MOV

**Reply by inspiredlabs** [reply]
When you scroll the content, it also invokes the horizontal slidy, see -2px @8 seconds on your vid. 

On the previous version, it doesn't do this at all, see line 40: `sensity: 6.25, // lower number is more sensitive` from the previous version of ['svelte-slidy'](https://svelte.dev/repl/5c149e1aeef242bd889a7ecf115c02b5?version=3.46.4).

**Reply by Valexr** [reply]
Ah… I see you point - you use scrollable area inside slide item…🤔
I watch the demo - same behaviour as v3 in mobile…🤷🏻‍♂️
It’s wasn’t prevented & not preventing now…
I need time for thinking about this case…🤓

**Reply by Valexr** [reply]
Oh, I remembered 😁 there's `gravity` props. A value of `gravity: 2` will completely stop the sliding. You can somehow hang a change of this props on the scroll inside the slide. Because it will be expensive to make an automatic preventive - you need to find out if it is inside the slide and then process it...

**Reply by Valexr** [reply]
And you need return `gravity` to previous value when you stop scroll content 👌🏻

**Reply by Valexr** [reply]
Something like [this](https://svelte.dev/repl/a46a0e9a9b514107921e6d96bf72aadb)

**Reply by inspiredlabs** [reply]
🙃 Whoa – sounds complicated! 

I have created a reduced test case in this REPL: https://svelte.dev/repl/d925cb5912d845d98f707327b3bfdb64?version=3.46.4

Please can you explain it further forking the demo above? I don't know what you mean.

**Reply by Valexr** [reply]
> Something like [this](https://svelte.dev/repl/a46a0e9a9b514107921e6d96bf72aadb)

I'l maked comments in code for explain ☝🏻

**Reply by Valexr** [reply]
1. Listen scroll event in scrollable slide item with setTimeout(500)
2. Set `gravity: 2` for stops sliding when content scrolling
3. When haven't scroll event & 500ms end on timer - set`gravity: 1.2` to default for activate sliding

That's all

**Reply by inspiredlabs** [reply]
Right 'o – thanks. If you think this is better, that's that then. 

Maybe if you do consider different approaches in the future, such as a `sensitivity` option, then that'd get a +1 from me.

Thanks again.

**Reply by Valexr** [reply]
You are not starred @slidy repo yet...?
🤔 i think its not ethic

**Comment by Valexr** [comment]
& thanks for issue respond - I'm forget making clamp behavior for wheel event 😁👍🏻

**Comment by Valexr** [comment, marked as answer]
Good news @inspiredlabs 🤓 [3.1.0 test draft](https://svelte.dev/repl/a46a0e9a9b514107921e6d96bf72aadb)
By default in `clamp: true` mode preventing crossscroll behavior✌🏻
Tx for the tip 🚀

p.s. release soon 😎
